"TypeError: object of type 'float' has no len()" for -inf entropy (single atom in ORCA)

The following comes from Travis:

```python3
    def testORCA_ORCA4_1_single_atom_freq_out(logfile):

        """ORCA frequency with single atom."""

        assert len(logfile.data.vibdisps) == 0

        assert len(logfile.data.vibfreqs) == 0

        assert len(logfile.data.vibirs) == 0

>       assert len(logfile.data.entropy) == -numpy.inf

E       TypeError: object of type 'float' has no len()

test/regression.py:1651: TypeError
```

I want to address it in #812. It is a simple fix, just summing the actual contributions for entropy.
(Giving `-inf` entropy on single atom calculations is [an issue on ORCA's side](https://orcaforum.kofo.mpg.de/viewtopic.php?f=8&t=5436).)

--- a/cclib/parser/orcaparser.py
+++ b/cclib/parser/orcaparser.py
@@ -780,7 +780,8 @@ class ORCA(logfileparser.Logfile):
             line = next(inputfile)
             while line[:17] != 'Electronic energy':
                 line = next(inputfile)
-            self.zpe = next(inputfile).split()[4]
+            self.electronic_energy = float(line.split()[3])
+            self.zpe = float(next(inputfile).split()[4])
             thermal_vibrational_correction = float(next(inputfile).split()[4])
             thermal_rotional_correction = float(next(inputfile).split()[4])
             thermal_translational_correction = float(next(inputfile).split()[4])
@@ -793,7 +794,11 @@ class ORCA(logfileparser.Logfile):
                 line = next(inputfile)
             thermal_enthalpy_correction = float(next(inputfile).split()[4])
             next(inputfile)
-            self.enthalpy = float(next(inputfile).split()[3])
+
+            if self.natom > 1:
+                self.enthalpy = float(next(inputfile).split()[3])
+            else:
+                self.enthalpy = self.electronic_energy + thermal_translational_correction
 
             # Entropy
             line = next(inputfile)
@@ -804,12 +809,20 @@ class ORCA(logfileparser.Logfile):
             rotational_entropy = float(next(inputfile).split()[3])
             translational_entropy = float(next(inputfile).split()[3])
             next(inputfile)
-            self.entropy = float(next(inputfile).split()[4])
+
+            if self.natom > 1:
+                self.entropy = float(next(inputfile).split()[4])
+            else:
+                self.entropy = (electronic_entropy + translational_entropy) / self.temperature
 
             line = next(inputfile)
             while (line[:25] != 'Final Gibbs free enthalpy') and (line[:23] != 'Final Gibbs free energy'):
                 line = next(inputfile)
-            self.freeenergy = float(line.split()[5])
+
+            if self.natom > 1:
+                self.freeenergy = float(line.split()[5])
+            else:
+                self.freeenergy = self.enthalpy - self.temperature * self.entropy
 
         # Read TDDFT information
         if any(x in line for x in ("TD-DFT/TDA EXCITED", "TD-DFT EXCITED")):

--- a/test/regression.py
+++ b/test/regression.py
@@ -1648,8 +1648,9 @@ def testORCA_ORCA4_1_single_atom_freq_out(logfile):
     assert len(logfile.data.vibdisps) == 0
     assert len(logfile.data.vibfreqs) == 0
     assert len(logfile.data.vibirs) == 0
-    assert len(logfile.data.entropy) == -numpy.inf
-    assert len(logfile.data.freeenergy) == numpy.inf
+    numpy.testing.assert_almost_equal(logfile.data.enthalpy, -460.14376, 5)
+    numpy.testing.assert_almost_equal(logfile.data.entropy, 6.056e-5, 8)
+    numpy.testing.assert_almost_equal(logfile.data.freeenergy, -460.16182, 6)
 
 
 def testORCA_ORCA4_2_MP2_gradient_out(logfile):

